### PR TITLE
Check histogram bin overflow in the support type, not the power type

### LINKED ISSUE #6408: Histograms should not overflow if bins fit into target type
One way to make histograms is to specify powers of 10, and have the internals generate 10 bins in each. The type we use to represent the power is different from the type of the underlying support, e.g., `u64` bins use a `u16` type for specifying the power. Here:

https://github.com/oxidecomputer/omicron/blob/d8f46b2d45f2939e5a41f18ad5176d15412b9d96/oximeter/types/src/histogram.rs#L1032-L1035

We are intending to check that the exponentiation we're being asked to do will fit in the target type. E.g., will `10 ** 10` be represented faithfully in a `u64`? The answer is "yes", but that's not what we're doing here. We're checking if `10u16 ** 10u16` will overflow, and we _want_ to check if `10u64 ** 10u16` will overflow. I even have a note about it, I just did the wrong thing.

### PR DESCRIPTION
**Pull request by bnaecker**
Fixes #6408

### CODE REVIEW

**Comment by bnaecker** [comment]
Friendly ping on this one @zeeshanlakhani, I'd like to build on this to get some other testing done. Thanks!

File: oximeter/types/src/histogram.rs [hunk 1]
```diff
@@ -1029,8 +1029,10 @@ where
             return Err(QuantizationError::InvalidSteps);
         }
 
-        // The highest power must be representable in the target type.
-        if self.checked_pow(hi.into()).is_none() {
+        // The highest power must be representable in the target type. Note that
+        // we have to convert to that target type _before_ doing this check.
+        let base = <u64 as From<Self>>::from(*self);
```
Comment (bnaecker) [inline oximeter/types/src/histogram.rs:1034]: This is not correct, it's checking in `u64`-land, needs to try to convert to the bin type `T`

Comment (bnaecker) [reply oximeter/types/src/histogram.rs:1034]: Fixed in d0ccdb0

**Review by zeeshanlakhani — APPROVED** [review]
Good find and fix.

**Pull request merged**